financial-connections/src/main/java/com/stripe/android/financialconnections/features/institutionpicker/InstitutionPickerViewModel.kt

```diff
@@ -88,14 +88,14 @@ internal class InstitutionPickerViewModel @Inject constructor(
 
     fun onQueryChanged(query: String) {
         searchJob += suspend {
-            delay(SEARCH_DEBOUNCE_MS)
-            val (result, millis) = measureTimeMillis {
-                searchInstitutions(
-                    clientSecret = configuration.financialConnectionsSessionClientSecret,
-                    query = query
-                )
-            }
-            if (query.isNotEmpty()) {
+            if (query.isNotBlank()) {
+                delay(SEARCH_DEBOUNCE_MS)
```

Comment:
why a `delay` is needed here?

Reply:
it debounces the search, so that if user types too fast we don't trigger unnecessary intermediate searches.